### DISCUSSION TITLE
How to add Http Only to fetchOption?

### DISCUSSION THREAD

**Discussion by softshipper** [urql-graphql/urql#1606] [Q&A] [answered]
Hi all  

I am using https://github.com/auth0/nextjs-auth0 to protect my NextJS web application and I followed the tutorial on https://formidable.com/open-source/urql/docs/advanced/server-side-rendering/#nextjs how to use URQL client on SSR. 
In the `getServerSideProps` I have the following implementation:

```
export const getServerSideProps = async () => {

const session = await getSession(req, res);
const ssrCache = ssrExchange({isClient: false});
const client = initUrqlClient({
    url: addr.gql,
    exchanges: [dedupExchange, cacheExchange, ssrCache, fetchExchange],
    fetchOptions: () => {
        ???
    }
}, false); 
```
Regarding to the post https://github.com/auth0/nextjs-auth0/issues/388#issuecomment-831802743 I have to pass on the request the session cookie, which will be validated on server side. The question is, how can I add the session cookie to the `fetchOptions`?  

Thanks

**Comment by kitten** [comment, marked as answer]
Hiya 👋

frankly, I wasn't quite sure how to answer this since you have seemingly a lot of stuff going on. If you manually need to add a cookie on the server-side you could forward it by adding `fetchOptions.header.Cookie`, to imitate it as your browser would send it
